### DISCUSSION TITLE
Docker Swarm Setup

### DISCUSSION THREAD

**Discussion by Snuffsis** [coollabsio/coolify#1846] [Q&A]
Hello.

So I have been looking at the documentation quite a bit and trying to find out a bit more about the docker swarm support (I am aware it is marked as experimental still), but I can't figure out how coolify fits in such a setup?

Should the instance running coolify also be the instance acting as the docker swarm manager, or should coolify act simply as something like a reverse proxy, a seperate machine where traffic comes into the network?

So, should it look something like this?
![image](https://github.com/coollabsio/coolify/assets/8822958/2eb1d9cf-cb87-42f1-b7f6-900c8e32673d)

Or should it look something like this?

![image](https://github.com/coollabsio/coolify/assets/8822958/d302fd6f-fb5c-4738-8cca-08b69ae15225)

One reason why I got these questions is that there aren't any mentions of actually turning the coolify instance into the docker swarm manager, and there is no option for changing the existing "localhost" server in the ui to be a docker swarm manager or docker swarm worker.

I would also assume that the docker swarm workers (and the manager?) shouldn't run coolify themselves.

**Comment by maietta** [comment]
Coolify is a deployment system. You don't have to run it on the same server. In fact, I'm running it on a Raspberry Pi and use it to help me deploy stuff elsewhere.

From what I understand, you can simply create a Docker Swarm like you normally would without Coolify. Then you add one of the manager nodes as your server and check the option indicating the server is part of a Docker Swarm.

I haven't verified this, but I think the Docker Swarm option tells Coolify to deploy as a stack rather than a service for this server when it is used. That's typically how that would work.

**Reply by Snuffsis** [reply]
Alright.
So, just a thought then.
We kind of decided that we want to just let coolify act as a load balancer using its traefik config as reverse proxy. But we couldn't manage to get it to work with subdomains and wildstar records to generate them automatically when using multiple servers that is behind the coolify servers.

Might be a bit different than my original question here though.

**Reply by pango07** [reply]
Did you ever get this figured out?
